### DISCUSSION TITLE
Nequip and pytorch 1.12.1

### DISCUSSION THREAD

**Discussion by schiotz** [mir-group/nequip#366] [Q&A] [answered]
Hi,

The installation instructions say 

> PyTorch >= 1.8, !=1.9, <=1.11.*. 

How important is the <=1.11.*? On a cluster I am using, there is only PyTorch 1.12.1 if I want a reasonably recent Python and scipy.

**Comment by Linux-cpp-lisp** [comment, marked as answer]
Hi @schiotz ,

Unfortunately, it can be quite important.  Major updates in the JIT backends starting in 1.12 may or may not have far ranging impacts on speed and correctness.  (We are not the only MLIP effort who has noticed this, so far I am not confident enough in the debugging that has been done to _recommend_ any other version.) 

If you choose to try PyTorch 1.12 or other versions !=1.11, please carefully check the sanity of your results and report any unexpected/strange behaviors _or the lack thereof_: it would be very helpful in our efforts to eventually migrate.  See, for example, https://github.com/mir-group/nequip/discussions/311#discussioncomment-5231630 for current reported issues.

Building PyTorch from source is also an option that can, on occasion, also improve performance.  Or installing new versions from binaries using, say, `conda` or `mamba`/

**Reply by schiotz** [reply]
Thank you!

I think we will use the old toolchain/Python installed with PyTorch 1.10.0, and then try to reproduce the results with 1.12.1.  I will then report back if it behaves similarly in the two cases.

/Jakob

**Reply by Linux-cpp-lisp** [reply]
Sounds good---just know that there are occasionally also issues with PyTorch 1.10 again due to upstream bugs, although those are never silent as far as we know.  If `conda` environments are possible, I would still recommend PyTorch 1.11 (note that you *do not* need to have system CUDA of the same version as PyTorch CUDA from `conda` unless you are building LAMMPS;  if you are building LAMMPS I am happy to share scripts for building PyTorch from source against system CUDA if that is helpful).

**Comment by tgmaxson** [comment]
Just to add some small bit of information, we don't see any issues on our end when using PyTorch 2.0.0 (from mamba list as follows)

```
pytorch                   2.0.0           py3.9_cuda11.8_cudnn8.7.0_0    pytorch
```

Seemingly no accuracy issues or performance weirdness.  No promises to others but this ended up being the easiest version to get working consistently and I have had zero issues with it.  We make relatively small models, maybe that has some influence.

**Reply by tgmaxson** [reply]
I should add, 2.0.1 did not work.  Specifically, this version of 2.0.0 seems to work well.

**Reply by Linux-cpp-lisp** [reply]
Thanks @tgmaxson , this is very useful information to have.  What in 2.0.1 didn't work, and which exact git commit of `nequip` are you on?

**Reply by tgmaxson** [reply]
I downloaded allegro/nequip on 7-27-2023 from develop, not sure which commit though exactly.  2.0.1 would crash randomly and LAMMPS wouldn't compile with allegro.  Not sure if I have any additional useful information in that regard
